src/org/traccar/events/MaintenanceEventHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2016 Anton Tananaev (anton@traccar.org)
+ * Copyright 2016 Andrey Kunitsyn (andrey@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.events;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.traccar.BaseEventHandler;
+import org.traccar.Context;
+import org.traccar.model.Device;
+import org.traccar.model.Event;
+import org.traccar.model.Position;
+
+public class MaintenanceEventHandler extends BaseEventHandler {
+
+    public static final String ATTRIBUTE_MAINTENANCE_START = "device.maintenanceStart";
+    public static final String ATTRIBUTE_MAINTENANCE_INTERVAL = "device.maintenanceInterval";
+
+    @Override
+    protected Collection<Event> analyzePosition(Position position) {
+        Device device = Context.getIdentityManager().getDeviceById(position.getDeviceId());
+        if (device == null || !Context.getIdentityManager().isLatestPosition(position)) {
+            return null;
+        }
+
+        double maintenanceInterval = Context.getDeviceManager()
+                .lookupServerDouble(device.getId(), ATTRIBUTE_MAINTENANCE_INTERVAL, 0);
```

Comment:
Not really related to this future, but maybe we should have just one set of "lookup" methods? And always pass parameter to it? Do you remember why we decided to have two sets?